src/registers/model_specific.rs

```diff
@@ -343,8 +353,13 @@ mod x86_64 {
         }
 
         /// Write a given virtual address to the KernelGsBase register.
+        ///
+        /// ## Safety
+        ///
+        /// The caller must ensure that this write operation has no unsafe side
+        /// effects, as the segment base address might be in use.
```

Comment:
```suggestion
        /// The caller must ensure that this write operation has no unsafe side
        /// effects, as the segment base address might be in use.
```

As mentioned in my previous comment, the `KernelGsBase` register is not the same as the base of the `GS` registers, so it cannot be "in use".

Reply:
Ah, that's what you meant. I changed it to “The caller must ensure that a future call to `GS::swap` has no unsafe side effects.”

Are you fine with that wording, or would you prefer a different wording?

Reply:
I can't come up with anything better of the top of my head. My impression is that the safety implications are closely tied to how a user actually uses `KernelGsBase`, so it's hard to provide generic guidelines and this is probably the best we can do.